### PR TITLE
Fix plots performance

### LINKED ISSUE #4084: [Plots] plot performance degradation in vue from angular
#### Summary
Plotting data (after receiving the data) takes longer in new Vue plots as it did in AngularJS plots for large data sets. UI becomes locked during plotting.

#### Expected vs Current Behavior
Vue plots should perform at least as well as legacy AngularJS plots

#### Impact Check List

- [ ] Data loss or misrepresented data?
- [x] Regression? Did this used to work or has it always been broken?
- [ ] Is there a workaround available?
- [x] Does this impact a critical component?
- [ ] Is this just a visual bug?

#### Steps to Reproduce

1. ensure time conductor is in fixed time mode
2. select any plot object with decently sized data
2. select a larger timeframe if more data needed
3. observe time to plot
4. expand plot to max size (maximize window size, collapse tree and inspector panes)
5. observe time to plot

* This was observed with identical build environments and data, but different software builds, where the vue plots lagged significantly with the high res plot.

#### Environment
* Open MCT Version: 
* Deployment Type: 
* OS:
* Browser:

#### Additional Information

Good idea to check vue reactivity. if data is somehow being reactified.

### PR DESCRIPTION
**Pull request by shefalijoshi**
- Separates plots series telemetry data storage from it's configuration so that Vue reactivity does not creep into the data and adversely affect performance
- Debounces window resize handler (correctly) so that too many requests aren't made to retrieve telemetry data.
- Removes event listener for beforeunload on destroy

Resolves #4084 

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?

### Author Checklist

* [x] Changes address original issue?
* [x] Unit tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue?

### CODE REVIEW

File: src/plugins/plot/MctPlot.vue [hunk 1]
```diff
@@ -239,18 +240,13 @@ export default {
     watch: {
         plotTickWidth(newTickWidth) {
             this.onTickWidthChange(newTickWidth, true);
-        },
-        gridLines(newGridLines) {
-            this.setGridLinesVisibility(newGridLines);
-        },
-        cursorGuide(newCursorGuide) {
-            this.setCursorGuideVisibility(newCursorGuide);
```
Comment (davetsay) [inline src/plugins/plot/MctPlot.vue:247]: is this related to this pr?

File: src/plugins/plot/MctPlot.vue [hunk 2]
```diff
@@ -927,14 +924,6 @@ export default {
...
         },
 
-        setCursorGuideVisibility(cursorGuide) {
-            this.cursorGuide = cursorGuide === true;
-        },
-
-        setGridLinesVisibility(gridLines) {
-            this.gridLines = gridLines === true;
-        },
-
```
Comment (davetsay) [inline src/plugins/plot/MctPlot.vue:937]: is this related to this pr?

File: src/ui/layout/BrowseBar.vue [hunk 3]
```diff
@@ -233,6 +233,7 @@ export default {
     },
     mounted: function () {
         document.addEventListener('click', this.closeViewAndSaveMenu);
+        this.promptUserbeforeNavigatingAway = this.promptUserbeforeNavigatingAway.bind(this);
         window.addEventListener('beforeunload', this.promptUserbeforeNavigatingAway);
```
Comment (davetsay) [inline src/ui/layout/BrowseBar.vue:237]: is this related to this pr?

**Review by davetsay — CHANGES_REQUESTED** [review]
looks good. just some questions on some code that might have snuck in. if they are good changes, should be good to go in.

**Review by akhenry — APPROVED** [review]
Looks good to me. From some basic profiling I did locally with SWGs it looks much more performant. Processor usage on a single SWG plot dropped from 70%-80% to 30%-40% which is awesome. Nice work!

**Comment by akhenry** [comment]
@davetsay I'll leave it with you to follow up on your review comments and merge this.

File: src/plugins/plot/configuration/configStore.js [hunk 4]
```diff
@@ -25,7 +25,10 @@ function ConfigStore() {
 
 ConfigStore.prototype.deleteStore = function (id) {
     if (this.store[id]) {
-        this.store[id].destroy();
```
[the remark this answers is not in the document]
Comment (shefalijoshi) [reply src/plugins/plot/configuration/configStore.js:28]: Yeah. Not all config stores have it.

**Review by davetsay — APPROVED** [review]
LGTM

## Reviewer Checklist
| | |
| --- | :---: |
| Changes appear to address issue? | Y |
| Appropriate unit tests included? | Y |
| Code style and in-line documentation are appropriate? | Y |
| Commit messages meet standards? | Y |
| Has associated issue been labelled `unverified`? | Y |

**Pull request merged**